assets/src/dashboard/components/infiniteScroller/index.js

```diff
@@ -21,82 +21,128 @@ import { __ } from '@wordpress/i18n';
 /**
  * External dependencies
  */
-import {
-  useState,
-  useEffect,
-  useLayoutEffect,
-  useCallback,
-  useRef,
-} from 'react';
+import { useEffect, useRef, useReducer } from 'react';
 import PropTypes from 'prop-types';
+import styled from 'styled-components';
+
+// TODO
+const ScrollMessage = styled.div`
+  margin: 20px 0;
+  text-align: center;
+  width: 100%;
+`;
+
+const STATE = {
+  loadable: 'loadable',
+  loading_internal: 'loading_internal',
+  loading_external: 'loading_external',
+  complete: 'complete',
+};
+
+const ACTION = {
+  ON_INTERNAL_LOAD: 'internal on load',
+  ON_EXTERNAL_LOAD: 'external on load',
+  ON_ALL_LOADED: 'all loaded',
+  ON_LOAD_SUCCESS: 'load success',
+};
+
+const machine = {
+  [STATE.loadable]: {
+    [ACTION.ON_INTERNAL_LOAD]: STATE.loading_internal,
+    [ACTION.ON_EXTERNAL_LOAD]: STATE.loading_external,
+  },
+  [STATE.loading_internal]: {
+    [ACTION.ON_ALL_LOADED]: STATE.complete,
+    [ACTION.ON_LOAD_SUCCESS]: STATE.loadable,
+  },
+  [STATE.loading_external]: {
+    [ACTION.ON_ALL_LOADED]: STATE.complete,
+    [ACTION.ON_LOAD_SUCCESS]: STATE.loadable,
+  },
+  [STATE.complete]: {
+    [ACTION.ON_LOAD_SUCCESS]: STATE.loadable,
+  },
```

Comment:
do we want to be able to go back to loadable from here?

I thought once we've loaded all the content we don't want this to be able to do anything

Reply:
Ahh I see. you need this because you're now always rendering the component and you need to be able to start the flow over if the filter changes.

I would just change the action name to be more representative of the action happening here. maybe something like:
```
    [ACTION.ON_CAN_LOAD_MORE]: STATE.loadable,
```

This will prevent you from needing to do this check:
```
    if (loadState === STATE.complete && canLoadMore) {
      dispatch(ACTION.ON_LOAD_SUCCESS);
    }
```
and just make it:
```
  if (canLoadMore) {
      dispatch(ACTION.ON_CAN_LOAD_MORE);
  }
```

Reply:
great idea! thank you!